src/libs/actions/Report.js

```diff
@@ -336,23 +336,14 @@ function removeOptimisticActions(reportID) {
 function updateReportWithNewAction(reportID, reportAction) {
     const newMaxSequenceNumber = reportAction.sequenceNumber;
     const isFromCurrentUser = reportAction.actorAccountID === currentUserAccountID;
-    const lastReadSequenceNumber = lastReadSequenceNumbers[reportID] || 0;
-
-    // When handling an action from the current users we can assume that their
-    // last read actionID has been updated in the server but not necessarily reflected
-    // locally so we must first update it and then calculate the unread (which should be 0)
-    if (isFromCurrentUser) {
-        setLocalLastRead(reportID, newMaxSequenceNumber);
-    }
-
+    let lastReadSequenceNumber = lastReadSequenceNumbers[reportID] || 0;
     const messageText = lodashGet(reportAction, ['message', 0, 'text'], '');
 
     // Always merge the reportID into Onyx
     // If the report doesn't exist in Onyx yet, then all the rest of the data will be filled out
     // by handleReportChanged
     const updatedReportObject = {
         reportID,
-        unreadActionCount: newMaxSequenceNumber - lastReadSequenceNumber,
```

Comment:
Add this back in here but reference 

```js
lastReadSequenceNumbers[reportID] || 0;
```

directly with a comment that the value may have been modified by `setLocalLastRead()`

Reply:
OK. I see. This would be easier to read.